test/fixtures/stdout-producer.js

```diff
@@ -0,0 +1,13 @@
+'use strict';
+
+const add = 'test\n';
+
+var str = '';
+for (var i = 0; i < 100000; i++) {
+  str += add;
+}
+
+str += 'hey\n';
+
+process.stdout.write(str);
+process.exit();
```

Comment:
Done, though I'm not sure of the specifics. I could take a dive into it if need be.